x-pack/metricbeat/module/aws/cloudwatch/cloudwatch.go

```diff
@@ -94,23 +97,33 @@ func (m *MetricSet) Fetch(report mb.ReporterV2) error {
 	startTime, endTime := aws.GetStartTimeEndTime(m.Period)
 
 	// Get listMetrics and namespacesTotal from configuration
-	listMetrics, namespacesTotal := readCloudwatchConfig(m.CloudwatchConfigs)
+	listMetrics, resourceTypes, namespaceResourceType := readCloudwatchConfig(m.CloudwatchConfigs)
```

Comment:
I think this might happen real soon. I'm planning to add one more config parameter 😂😂 I will merge this PR and do some serious refactoring in the new PR.

Reply:
This piece of code is refactored by https://github.com/elastic/beats/pull/12840 😬